client/src/components/pages/Profile.js

```diff
@@ -1,7 +1,167 @@
+  const [profilePicture, setProfilePicture] = useState(null);
+  const [selectedFile, setSelectedFile] = useState(null);
+  const [error, setError] = useState("");
+  const [successMessage, setSuccessMessage] = useState("");
+  const [showMessage, setShowMessage] = useState(false);
+  const [animationClass, setAnimationClass] = useState("animate__fadeInUp");
+
+  const fadeTimeout = useRef(null);
+  const { token } = useToken();
+
+  const fadeCycle = () => {
+    fadeTimeout.current = setTimeout(() => {
+      setAnimationClass("animate__fadeOutUp");
+
+      fadeTimeout.current = setTimeout(() => {
+        setShowMessage(false);
+        setAnimationClass("animate__fadeInUp");
+      }, 500);
+    }, 3000);
+  };
+
+  const handleFetchProfilePicture = async () => {
+    clearTimeout(fadeTimeout.current);
+    setError("");
+    setSuccessMessage("");
+    setShowMessage(false);
+  
+    try {
+      if (!token) {
+        const authError = {
+          status: 401,
+          message: "User is not authenticated. Please log in.",
+        };
+        throw authError; 
+      }
+  
+      const decodedToken = jwtDecode(token);
+      const userId = decodedToken.sub; 
+  
+      if (!userId) {
+        throw new Error("Invalid token: User ID is missing.");
+      }
+
+      const profileUrl = await fetchProfilePicture(userId);
+      setProfilePicture(profileUrl);
+      setSuccessMessage("Profile picture fetched successfully!");
+      setShowMessage(true);
+      fadeCycle();
+    } catch (error) {
+      const errorMsg = error.response?.data?.message || error.message || "An error occurred.";
+      setError(`Failed to fetch profile picture: ${errorMsg}`);
+      setShowMessage(true);
+      fadeCycle();
+    }
+  };
+  
+
+  const handleUploadProfilePicture = async () => {
+    if (!selectedFile) {
+      setError("Please select a file before uploading.");
+      setShowMessage(true);
+      fadeCycle();
+      return;
+    }
+
+    clearTimeout(fadeTimeout.current);
+    setError("");
+    setSuccessMessage("");
+    setShowMessage(false);
+
+    try {
+      if (!token) {
+        const authError = {
+          status: 401,
+          message: "User is not authenticated. Please log in.",
+        };
+        throw authError; 
+      }
+      const profileUrl = await uploadProfilePicture(token, selectedFile);
+      setProfilePicture(profileUrl);
+      setSuccessMessage("Profile picture updated successfully!");
+      setShowMessage(true);
+      fadeCycle();
+    } catch (error) {
+      const errorMsg = error.response?.data?.message || error.message || "An error occurred.";
+      setError(`Failed to upload profile picture: ${errorMsg}`);
+      setShowMessage(true);
+      fadeCycle();
+    }
+  };
+
+  const handleDeleteProfilePicture = async () => {
+    clearTimeout(fadeTimeout.current);
+    setError("");
+    setSuccessMessage("");
+    setShowMessage(false);
+
+    try {
+      await deleteProfilePicture(token);
```

Comment:
To jest moja preferencja tylko, ale zamiast await uzylbym konstrukcji z .then() w przypadku zapytań na controller. Wtedy nie trzeba robić asynchronicznej funkcji HandleDeleteProfilePicture i mam wrażenie że kod jest wtedy trochę czytelniejszy dzięki użyciu .then(). Przykład jak takie coś 
może wyglądać jest w Login.js.